wordpress/wp-content/plugins/memberful-wp/src/user/map.php

```diff
@@ -19,6 +19,23 @@ class Memberful_User_Map {
    * @return WP_User
    */
   public function map( $member, array $context = array() ) {
+    if (isset($member->deleted)) {
+      $mapping_from_member = $this->repository()->find_user_member_is_mapped_to($member);
+
+      if ($mapping_from_member['mapping_exists']) {
+        return $mapping_from_member['user'];
+      } else {
+        return $this->add_data_to_wp_error(
+          new WP_Error(
+            'user_does_not_exist',
+            'The Wordpress user with the given member_id does not exist',
+            array('member_id' => $member->id)
+          ),
+          compact('member')
+        );
+      }
```

Comment:
Not blocking but I think we could extract this to a function to make it easier to read the rest of `map()`:
```php
if (isset($member->deleted)) {
  return $this->map_deleted_member($member);
}

...
```

Reply:
I like the idea, @lmansur ! I extracted the function in this commit: https://github.com/memberful/memberful-wp/pull/352/commits/4547fa46cf7626baafca5e1911f9d10944082a8b. I think it looks better now!